crates/bevy_render/src/color.rs

```diff
@@ -38,6 +38,43 @@ impl Color {
         Color { r, g, b, a }
     }
 
+    pub fn hex<T: AsRef<str>>(hex: T) -> Color {
+        let hex = hex.as_ref();
+
+        // RGB
+        if hex.len() == 3 {
+            let mut data = [0; 6];
+            for (i, ch) in hex.chars().enumerate() {
+                data[i * 2] = ch as u8;
+                data[i * 2 + 1] = ch as u8;
+            }
+            return decode_rgb(&data);
+        }
+
+        // RGBA
+        if hex.len() == 4 {
+            let mut data = [0; 8];
+            for (i, ch) in hex.chars().enumerate() {
+                data[i * 2] = ch as u8;
+                data[i * 2 + 1] = ch as u8;
+            }
+            return decode_rgba(&data);
+        }
+
+        // RRGGBB
+        if hex.len() == 6 {
+            return decode_rgb(hex.as_bytes());
+        }
+
+        // RRGGBBAA
+        if hex.len() == 8 {
+            return decode_rgba(hex.as_bytes());
+        }
+
+        // Invalid value, use default color
+        Color::default()
```

Comment:
WDYT about making this return `Result` rather than having a default? That feels a bit more flexible to me because the caller can choose their own default.

Reply:
Yeah I guess it's more rusty to return a result when something is being parsed. That also allows us to disambiguate between failures and default values.

Reply:
Even if there is an error, the user will find it immediately (because the color is not effective). If we return a `Result`, then we will use more `.unwrap()`.

Just like `Color::rgb(123456789.0, ...)` and `Color::hex("abcdefg")`, I think they are equivalent, but we can’t display this color, they will not panic.

Reply:
I think it's reasonable to return the error to the user if the hex can't be parsed. What if this string value comes from outside the application (like a user entered value, or something from the network).

I think it would be a nicer interface to give the application the chance to respond to the error, rather than silently changing the behavior of the program.

While I love that the API of `bevy` is simple, I think returning `Result` is a very rusty thing to do, and doesn't significantly increase the complexity of the API.

If a user wants the default color they can always do `Color::hex("abcdefg").unwrap_or(Color::default())`

Reply:
Awesome, looks great!